config/src/test/java/org/springframework/security/config/annotation/web/configurers/saml2/Saml2LogoutConfigurerTests.java

```diff
@@ -371,7 +371,7 @@ public void saml2LogoutRequestWhenLowercaseEncodingAndDifferentQueryParamOrderTh
 	}
 
 	@Test
-	public void saml2LogoutRequestWhenNoRegistrationThen400() throws Exception {
+	public void saml2LogoutRequestWhenNoRegistrationThen401() throws Exception {
```

Comment:
I'm not understanding the logic of this change. Can you please elaborate? Ideally, we leave tests as they are unless it is a bug.

Reply:
This was/is due to this, which I do think was implemented correctly
https://github.com/spring-projects/spring-security/pull/14676/files#r1533081906

and then there is also this https://github.com/spring-projects/spring-security/blob/84fc5a70eebe10066bc4139fbfb55e78aadfbdde/config/src/test/java/org/springframework/security/config/http/Saml2LogoutBeanDefinitionParserTests.java#L273

So I believe either one of these tests has to change and I'd argue that having an incorrect RegistrationID should result in a 401 rather than a 400